spec/reporter_spec.rb

```diff
@@ -0,0 +1,16 @@
+require_relative '../lib/forspell/reporter'
+require_relative '../lib/forspell/loaders/base'
+
+RSpec.describe Forspell::Reporter do
+
+  let(:reporter) { described_class.new(logfile: STDOUT, verbose: false, format: 'readable') }
+  let(:word) { Forspell::Loaders::Word.new('file.rb', 5, 'typo') }
+  let(:error) { [word, ['type']] }
+
+  describe '#error' do
+    it 'should add error with suggestions' do
+      reporter.error(*error)
+      expect(reporter.instance_variable_get(:@errors).size).to eq(1)
+    end
+  end
+end
```

Comment:
This doesn't seem to test anything that reporter really does :woman_shrugging: 
I'd say the easiest way to do it is to have several contexts with "before"  that setups something (file, error and so on), and then test "what it outputs".
(The tests "when I do something, some internal variables change" are generally to be avoided, it breaks incapsulation seriously and says nothing about "how the class works")

Reply:
Whoops.
I didn't know how to include `puts` testing, but after some googling it seems to be like `STDOUT.should_receive(:puts).with(...)`.

Reply:
`expect { block }.to output('something').to_stdout` :)